### DISCUSSION TITLE
Stranger mistake when compiling

### DISCUSSION THREAD

**Discussion by Vidorlp** [mriscoc/Ender3V2S1#547] [Q&A]
Sorry for my evil English, I am Spanish and I use a translator. I am trying to compile the last version with BL but when I compile this notice in yellow. I have compiled other times and I have not had problems.

PS C:\Marlin\Ender 3 V2S1 mriscoc> platformio run --target clean --silent -e STM32F103RE_creality
PS C:\Marlin\Ender 3 V2S1 mriscoc> echo "done" >"C:\Users\ea8cx\AppData\Local\Temp\ipc"
PS C:\Marlin\Ender 3 V2S1 mriscoc> platformio run --silent -e STM32F103RE_creality
In file included from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/WString.h:29,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/Print.h:27,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/Stream.h:26,        
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/HardwareSerial.h:29,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/WSerial.h:5, 
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/wiring.h:47, 
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/Arduino.h:36,
                 from c:\marlin\ender 3 v2s1 mriscoc\marlin\src\hal\shared\marduino.h:36,
                 from c:\marlin\ender 3 v2s1 mriscoc\marlin\src\hal\stm32\hal.h:28,
                 from Marlin\src\lcd\e3v2\proui\../../../inc/../HAL/HAL.h:30,
                 from Marlin\src\lcd\e3v2\proui\../../../inc/MarlinConfig.h:31,
                 from Marlin\src\lcd\e3v2\proui\../../../MarlinCore.h:24,
                 from Marlin\src\lcd\e3v2\proui\printstats.cpp:29:
Marlin\src\lcd\e3v2\proui\printstats.cpp: In static member function 'static void PrintStatsClass::Draw()':
Marlin\src\lcd\e3v2\proui\../../../inc/../HAL/HAL.h:41:17: warning: '%s' directive writing up to 29 bytes into a region of size 28 [-Wformat-overflow=]    
   41 |   #define S_FMT "%s"
C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/avr/pgmspace.h:71:40: note: in definition of macro 'sprintf_P'   
   71 | #define sprintf_P(s, ...) sprintf((s), __VA_ARGS__)
      |                                        ^~~~~~~~~~~
Marlin\src\lcd\e3v2\proui\printstats.cpp:55:18: note: in expansion of macro 'PSTR'
   55 |   sprintf_P(buf, PSTR(S_FMT ": %s"), GET_TEXT(MSG_INFO_PRINT_TIME), str);
      |                  ^~~~
Marlin\src\lcd\e3v2\proui\printstats.cpp:55:23: note: in expansion of macro 'S_FMT'
   55 |   sprintf_P(buf, PSTR(S_FMT ": %s"), GET_TEXT(MSG_INFO_PRINT_TIME), str);
      |                       ^~~~~
Marlin\src\lcd\e3v2\proui\printstats.cpp:60:32: note: format string is defined here
   60 |   sprintf_P(buf, PSTR(S_FMT ": %s m"), GET_TEXT(MSG_INFO_PRINT_FILAMENT), dtostrf(ps.filamentUsed / 1000, 1, 2, str));
      |                                ^~
In file included from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/WString.h:29,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/Print.h:27,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/Stream.h:26,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/HardwareSerial.h:29,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/WSerial.h:5,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/wiring.h:47,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/Arduino.h:36,
                 from c:\marlin\ender 3 v2s1 mriscoc\marlin\src\hal\shared\marduino.h:36,
                 from c:\marlin\ender 3 v2s1 mriscoc\marlin\src\hal\stm32\hal.h:28,
                 from Marlin\src\lcd\e3v2\proui\../../../inc/../HAL/HAL.h:30,
                 from Marlin\src\lcd\e3v2\proui\../../../inc/MarlinConfig.h:31,
                 from Marlin\src\lcd\e3v2\proui\../../../MarlinCore.h:24,
                 from Marlin\src\lcd\e3v2\proui\printstats.cpp:29:
C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/avr/pgmspace.h:71:34: note: 'sprintf' output between 23 and 52 bytes into a destination of size 50
   71 | #define sprintf_P(s, ...) sprintf((s), __VA_ARGS__)
      |                           ~~~~~~~^~~~~~~~~~~~~~~~~~
Marlin\src\lcd\e3v2\proui\printstats.cpp:55:3: note: in expansion of macro 'sprintf_P'
   55 |   sprintf_P(buf, PSTR(S_FMT ": %s"), GET_TEXT(MSG_INFO_PRINT_TIME), str);
      |   ^~~~~~~~~
Marlin\src\lcd\e3v2\proui\../../../inc/../HAL/HAL.h:41:17: warning: '%s' directive writing up to 29 bytes into a region of size 27 [-Wformat-overflow=]    
   41 |   #define S_FMT "%s"
C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/avr/pgmspace.h:71:40: note: in definition of macro 'sprintf_P'
   71 | #define sprintf_P(s, ...) sprintf((s), __VA_ARGS__)
      |                                        ^~~~~~~~~~~
Marlin\src\lcd\e3v2\proui\printstats.cpp:58:18: note: in expansion of macro 'PSTR'
   58 |   sprintf_P(buf, PSTR(S_FMT ": %s"), GET_TEXT(MSG_INFO_PRINT_LONGEST), str);
      |                  ^~~~
Marlin\src\lcd\e3v2\proui\printstats.cpp:58:23: note: in expansion of macro 'S_FMT'
   58 |   sprintf_P(buf, PSTR(S_FMT ": %s"), GET_TEXT(MSG_INFO_PRINT_LONGEST), str);
      |                       ^~~~~
Marlin\src\lcd\e3v2\proui\printstats.cpp:60:32: note: format string is defined here
   60 |   sprintf_P(buf, PSTR(S_FMT ": %s m"), GET_TEXT(MSG_INFO_PRINT_FILAMENT), dtostrf(ps.filamentUsed / 1000, 1, 2, str));
      |                                ^~
In file included from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/WString.h:29,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/Print.h:27,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/Stream.h:26,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/HardwareSerial.h:29,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/wiring.h:47,
                 from C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/Arduino.h:36,
                 from c:\marlin\ender 3 v2s1 mriscoc\marlin\src\hal\shared\marduino.h:36,
                 from c:\marlin\ender 3 v2s1 mriscoc\marlin\src\hal\stm32\hal.h:28,
                 from Marlin\src\lcd\e3v2\proui\../../../inc/../HAL/HAL.h:30,
                 from Marlin\src\lcd\e3v2\proui\../../../inc/MarlinConfig.h:31,
                 from Marlin\src\lcd\e3v2\proui\../../../MarlinCore.h:24,
                 from Marlin\src\lcd\e3v2\proui\printstats.cpp:29:
C:\Users\ea8cx\.platformio\packages\framework-arduinoststm32@4.10900.200819\cores\arduino/avr/pgmspace.h:71:34: note: 'sprintf' output between 24 and 53 bytes into a destination of size 50
   71 | #define sprintf_P(s, ...) sprintf((s), __VA_ARGS__)
      |                           ~~~~~~~^~~~~~~~~~~~~~~~~~
Marlin\src\lcd\e3v2\proui\printstats.cpp:58:3: note: in expansion of macro 'sprintf_P'
   58 |   sprintf_P(buf, PSTR(S_FMT ": %s"), GET_TEXT(MSG_INFO_PRINT_LONGEST), str);
      |   ^~~~~~~~~
PS C:\Marlin\Ender 3 V2S1 mriscoc> echo "done" >"C:\Users\ea8cx\AppData\Local\Temp\ipc"
PS C:\Marlin\Ender 3 V2S1 mriscoc>

**Comment by mriscoc** [comment]
Some Spanish strings are longer than English ones, you need increase the size of some char buffers to allocate them.

**Reply by Vidorlp** [reply]
I have sought information on how to do it and I found these instructions but I still give me the same error.


    Go to menu File → Preferences → Settings.
    Search for scrollback.
    Overwrite "terminal.integrated.scrollback": 1000, (found in the left panel) with whatever you want in the right panel, e.g. 1000000.

**Reply by mriscoc** [reply]
You need to know about C++ programming.

**Reply by Vidorlp** [reply]
And why does it happen with this version and not others?

**Reply by mriscoc** [reply]
That can happen with any memory optimized c++ program.

**Reply by Vidorlp** [reply]
Thank you very much, we will use English then.
